employees/tests/test_unit_custom_forms.py

```diff
@@ -9,28 +9,37 @@
 from employees.forms import DurationFieldForm
 from employees.forms import MonthSwitchForm
 from employees.forms import ProjectJoinForm
+from managers.factories import ProjectFactory
 from managers.models import Project
 from users.factories import UserFactory
 
 
 class ProjectJoinFormTests(TestCase):
+    def setUp(self) -> None:
+        self.queryset_length = 10
+        self._generate_test_projects(self.queryset_length)
+        self.user = UserFactory()
+        self.project_queryset = Project.objects.all().order_by("name")
+
     def test_project_join_form_should_create_choice_field_with_project_name_and_id_based_author_passed_in_initial(self):
-        queryset_length = 10
-        for i in range(queryset_length):
-            project = Project(name=f"Test Project {i}", start_date=datetime.datetime.now())
-            project.save()
-
-        custom_user = UserFactory()
-        custom_kwargs = {"initial": {"author": custom_user}}
-        queryset = Project.objects.all()
+        custom_kwargs = {"initial": {"author": self.user}}
         form = ProjectJoinForm(**custom_kwargs)
         choices = form.fields["projects"].choices
 
         self.assertIsNotNone(choices)
-        self.assertEqual(len(choices), queryset_length)
-        for i in range(queryset_length):
-            self.assertEqual(choices[i][0], queryset[i].id)
-            self.assertEqual(choices[i][1], queryset[i].name)
+        self.assertEqual(len(choices), self.queryset_length)
+
+        self._compare_id_and_names_of_choices_and_project_for_projects(choices)
+
+    @staticmethod
+    def _generate_test_projects(queryset_length: int):
+        for _project_number in range(queryset_length):
```

Comment:
You can just use here `_` instead of `_project_number`

Reply:
Done